modules/pytest/Makefile

```diff
@@ -1,14 +1,50 @@
-pytest: pytest/test
+PYTEST = $(PIPENV_RUN) pytest
+
+.PHONY: pytest/init
+pytest/init: pipenv reports/
+
 .PHONY: pytest
+## Run Python tests with pytest
+pytest: pytest/init ; $(PYTEST)
 
-pytest/test: pipenv reports/
-	$(PIPENV_RUN) pytest -m "not postbuild"
-.PHONY: pytest/test
+.PHONY: pytest/lf
+pytest/lf: pytest/init ; $(PYTEST) --lf
+
+.PHONY: pytest/pdb
+pytest/pdb: pytest/init ; $(PYTEST) --pdb
+
+.PHONY: pytest/mf
+pytest/mf/%: pytest/init ; $(PYTEST) --maxfail=$*
+
+.PHONY: pytest/k/%
+pytest/k/%: pytest/init ; $(PYTEST) -k "$*"
+
+.PHONY: pytest/m/%
+pytest/m/%: pytest/init ; $(PYTEST) -m "$*"
+
+####
+# Targets calling specific custom flags we commonly see used.
+####
+
+.PHONY: pytest/pre-build
+pytest/pre-build: ; @$(MAKE) pytest/m/"not postbuild"
+
+.PHONY: pytest/post-build
+pytest/post-build: ; @$(MAKE) pytest/m/"postbuild"
+
+####
+# Deprecated targets.
+####
+
+define pytest-deprecated
+	@echo "WARNING: This target is being deprecated. Please switch from '$(1)' to '$(2)'"
```

Comment:
nit: `deprecation` already refers to the process of future replacement, so I think it's better to say `This target is deprecated.`

Reply:
Fixed